src/core/components/bitswap.js

```diff
@@ -16,16 +18,27 @@ module.exports = function bitswap (self) {
       const list = self._bitswap.getWantlist()
       return formatWantlist(list)
     },
-    stat: () => {
+
+    stat: promisify((callback) => {
       if (!self.isOnline()) {
-        throw new Error(OFFLINE_ERROR)
+        callback(new Error(OFFLINE_ERROR))
```

Comment:
This needs two things:

- Be wrapped with a setImmediate from the `async` module
- Have a `return`, otherwise the rest of the function will be executed.